src/Automation/Interfaces/IScanner.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+
+namespace Axe.Windows.Automation
+{
+    /// <summary>
+    /// Runs AxeWindows automated tests
+    /// </summary>
+    public interface IScanner
+    {
+        /// <summary>
+        /// Run AxeWindows automated tests
+        /// </summary>
+        /// <remarks>
+        /// If a value was provided in <see cref="Config.OutputDirectory"/>,
+        /// output files may be written to the specified directory.
+        /// (Note: no output files will be written if no errors were found.)
+        /// An exception may be thrown if the value of <see cref="Config.ProcessId"/> is invalid
+        /// or if the directory provided in <see cref="Config.OutputDirectory"/> does not exist.
```

Comment:
>if the directory provided in <see cref="Config.OutputDirectory"/> does not exist [](start = 15, length = 80)

Is this the final design? You seemed to agree when I suggested that auto-creation of an output folder is a common thing in test tools

Reply:
I do agree. I just didn't remember there was a comment specifically calling this out.

Reply:
In email, IIRC

---
In reply to: [291298458](https://github.com/microsoft/axe-windows/pull/46#discussion_r291298458) [](ancestors = 291298458)